src/services/actions-service/actions-service.ts

```diff
@@ -157,4 +157,26 @@ class ActionsService implements IActionsService {
     }
     return actions;
   }
+
+  sortActions(actions: Action[]): Action[] {
+    return actions.sort((a, b) => {
```

Comment:
Since actions are returned grouped by block, this is focused on ordering the actions within a block.

I'm not sure the sort order is correct.  Maybe it's reversed,  but we should get it right, then we will have access to the unit test as a spec.

Reply:
@deepthiskumar it's worth noting that you and gregor both mentioned a sub-order to the account update in https://github.com/o1-labs/o1js/issues/1872

It is your point 4, and Gregor said "It's extremely important that when reordering the actions here, the order within the same account update is maintained".

This sort only looks at sequence number and account update id.  But with my JSON comment on this PR and this code, we should be able to get on the same page more easily.  If there's a third sorting lever we need here, then I can add it, but it's not clear to me what it is.

Reply:
Yes, the third sorting lever is the element ID index from `zkappEventElementIds`

Reply:
So as linked in my [other comment](https://github.com/o1-labs/Archive-Node-API/pull/107/files#r1882998656), I think we only need to sort to the `event field array` level.  Then all the fields from all the field arrays get put into `data` in the correct order.